Schema Statements.sql

```diff
@@ -0,0 +1,135 @@
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Dumping data for table `category`
+--
+
+LOCK TABLES `category` WRITE;
+/*!40000 ALTER TABLE `category` DISABLE KEYS */;
+INSERT INTO `category` VALUES (0,'foods'),(1,'entertainment'),(2,'misc');
+/*!40000 ALTER TABLE `category` ENABLE KEYS */;
+UNLOCK TABLES;
+
+--
+-- Table structure for table `items`
+--
+
+DROP TABLE IF EXISTS `items`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `items` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `name` varchar(45) DEFAULT NULL,
+  `origin_id` int NOT NULL,
+  `price` decimal(19,2) DEFAULT NULL,
+  `fk_category` int DEFAULT NULL,
+  `fk_qtyType` int DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  KEY `fk_category_idx` (`fk_category`),
+  KEY `fk_qtyType_idx` (`fk_qtyType`),
+  KEY `origin_id_idx` (`origin_id`),
+  CONSTRAINT `fk_category` FOREIGN KEY (`fk_category`) REFERENCES `category` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT,
+  CONSTRAINT `fk_qtyType` FOREIGN KEY (`fk_qtyType`) REFERENCES `qty_type` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT,
+  CONSTRAINT `origin_id` FOREIGN KEY (`origin_id`) REFERENCES `origin` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT
+) ENGINE=InnoDB AUTO_INCREMENT=8 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Dumping data for table `items`
+--
+
+LOCK TABLES `items` WRITE;
+/*!40000 ALTER TABLE `items` DISABLE KEYS */;
+INSERT INTO `items` VALUES (7,'bob',2,300.00,0,0);
+/*!40000 ALTER TABLE `items` ENABLE KEYS */;
+UNLOCK TABLES;
+
+--
+-- Table structure for table `origin`
+--
+
+DROP TABLE IF EXISTS `origin`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `origin` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `name` varchar(45) DEFAULT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB AUTO_INCREMENT=5 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Dumping data for table `origin`
+--
+
+LOCK TABLES `origin` WRITE;
+/*!40000 ALTER TABLE `origin` DISABLE KEYS */;
+INSERT INTO `origin` VALUES (1,'Woolworths'),(2,'Aldi'),(4,'Coles');
+/*!40000 ALTER TABLE `origin` ENABLE KEYS */;
+UNLOCK TABLES;
+
+--
+-- Table structure for table `qty_type`
+--
+
+DROP TABLE IF EXISTS `qty_type`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `qty_type` (
+  `id` int NOT NULL,
+  `type` varchar(45) DEFAULT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Dumping data for table `qty_type`
+--
+
+LOCK TABLES `qty_type` WRITE;
+/*!40000 ALTER TABLE `qty_type` DISABLE KEYS */;
+INSERT INTO `qty_type` VALUES (0,'G'),(1,'L'),(2,'each');
```

Comment:
Grams are in lowercase (`g` not `G`).
But regardless I think such names will be used mostly internally (we can make things pretty on front end), so we should stick with consistent all caps or all lower.

Reply:
Also should probably be ml not l, so that we can allow for quantity to be an integer.